altsrc/map_input_source.go

```diff
@@ -127,19 +128,46 @@ func (fsm *MapInputSource) Float64(name string) (float64, error) {
 	return 0, nil
 }
 
+func castToInt64(v interface{}) (int64, bool) {
+	int64Value := int64(0)
+	isType := false
+	if v == nil {
+		return int64Value, true
+	}
+
+	// There are only four cases(int, int64, uint64, float64) when parsing the integer in yaml.v3 pkg
+	// But the last case, float64, is an error case so that ignored
+	vType := reflect.TypeOf(v).Kind()
+	switch vType {
+	case reflect.Int:
+		int64Value = int64(v.(int))
+		isType = true
+	case reflect.Int64:
+		int64Value = v.(int64)
+		isType = true
+	case reflect.Uint64:
```

Comment:
if user expects an int64 but a uint64 is provided should we still convert ? Discuss

Reply:
I agree. It is better to occur an error.